### PR TITLE
Patroni updates

### LINKED ISSUE #17: pgsql_patroni statefulset updates
Improvements for the statefulset
- add anti-affinity rules to statefulset 
- look into imagepullpolicy improvement

### PR DESCRIPTION
**Pull request by cvarjao**
Fix #17:

- Add readiness probe
- Add build/deployment templates
- Add tests
- Enable patroni configuration file to be configurable using environment variables
- Add post init script for creating user/database for the app

### CODE REVIEW

File: apps/pgsql/patroni/openshift/deployment-prereq.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  annotations:
+    description: |-
+      Patroni Postgresql database cluster, with persistent storage.
+    iconClass: icon-postgresql
+    openshift.io/display-name: Patroni Postgresql (Persistent)
+    openshift.io/long-description: This template deploys a a patroni postgresql HA
```
Comment (jefkel) [inline apps/pgsql/patroni/openshift/deployment-prereq.yaml]: Remove extra "a"

File: apps/pgsql/patroni/openshift/deployment-prereq.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      cluster without persistent storage.
```
Comment (jefkel) [inline apps/pgsql/patroni/openshift/deployment-prereq.yaml]: change "without" to "with"

File: apps/pgsql/patroni/openshift/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,321 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  annotations:
+    description: |-
+      Patroni Postgresql database cluster, with persistent storage.
+    iconClass: icon-postgresql
+    openshift.io/display-name: Patroni Postgresql (Persistent)
+    openshift.io/long-description: This template deploys a a patroni postgresql HA
+      cluster without persistent storage.
```
Comment (jefkel) [inline apps/pgsql/patroni/openshift/deployment.yaml]:
```suggestion
      cluster with persistent storage.
```

File: apps/pgsql/patroni/openshift/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,321 @@
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  annotations:
+    description: |-
+      Patroni Postgresql database cluster, with persistent storage.
+    iconClass: icon-postgresql
+    openshift.io/display-name: Patroni Postgresql (Persistent)
+    openshift.io/long-description: This template deploys a a patroni postgresql HA
```
Comment (jefkel) [inline apps/pgsql/patroni/openshift/deployment.yaml]:
```suggestion
    openshift.io/long-description: This template deploys a patroni postgresql HA
```

File: apps/pgsql/patroni/openshift/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+  value: patroni:10
+- description: The size of the persistent volume to create.
+  displayName: Persistent Volume Size
+  name: PVC_SIZE
+  value: 1Gi
+- name: STORAGE_CLASS
+  value: gluster-block
+- name: OPENSHIFT_IMAGE_REGISTRY
+  #docker-registry.default.svc:5000
+  value: 172.30.1.1:5000
```
Comment (jefkel) [inline apps/pgsql/patroni/openshift/deployment.yaml]:
```suggestion
  value: docker-registry.default.svc:5000
```

**Comment by stewartshea** [comment]
@cvarjao is this PR waiting on anything else?

**Comment by cvarjao** [comment]
@stewartshea , it should be good to go now after my last commit (2c270e6), please use "squash  and merge" when accepting it.

**Pull request merged**